### PR TITLE
Revert "[LLVMGPU] Deprecate the matmul simt pipeline (#19335)"

### PR DESCRIPTION
**Pull request by archana-ramalingam**
This reverts commit 6ff00a8a008d06b604d4ca4e0ae6e601ae810b4f.
The above commit causes Llama3.1 8B fp16 model to generate NaN logits for prefill/decode. 
Issue: https://github.com/iree-org/iree/issues/19506

### CODE REVIEW

**Comment by MaheshRavishankar** [comment]
Please link the issue here

**Review by MaheshRavishankar — APPROVED** [review]

**Review by pashu123 — APPROVED** [review]

**Comment by MaheshRavishankar** [comment]
Cc @pashu123

**Comment by pashu123** [comment]
> Cc @pashu123

Taking a look.

**Comment by ScottTodd** [comment]
Re-triggered all the workflows that failed. Errors looked like transient network issues on GitHub's side.
